### PR TITLE
[Java][jersey3] Add error entity deserialization to ApiException

### LINKED ISSUE #4777: [BUG] ApiException does not contains Error object
##### Description
After generating Java client, in case of error, ApiException is thrown but it doesn't contains the Error object defined in specs file.

##### openapi-generator version
release : 4.2.2

##### OpenAPI declaration file content or url
Below my spec file where Error object should be returned in case http status code is not 200.

```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: petstore
  description: |
    This is the specification of Petstore with the following resources:

    - Pet
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      tags:
        - pets
      parameters:
        - name: limit
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            type: integer
            format: int32
      responses:
        '200':
          description: A paged array of pets
          headers:
            x-next:
              description: A link to the next page of responses
              schema:
                type: string
          content:
            application/json:    
              schema:
                $ref: "#/components/schemas/Pets"
        default:
          description: unexpected error
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Error"
    
components:
  schemas:
    Pet:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
        tag:
          type: string
    Pets:
      type: array
      items:
        $ref: "#/components/schemas/Pet"
    Error:
      type: object
      required:
        - code
        - message
      properties:
        code:
          type: integer
          format: int32
        message:
          type: string

```

##### Command line used for generation
```bash
./gradlew clean openApiGenerate
```
with the following `build.gradle` file:
```groovy
plugins {
  id "org.openapi.generator" version "4.2.2"
  id 'java'
}

repositories {
  mavenCentral()
}


dependencies {
    compile 'com.squareup.okhttp3:okhttp:4.2.2'
    compile 'com.squareup.okhttp3:logging-interceptor:4.2.2'
    compile 'org.threeten:threetenbp:1.4.0'
    compile 'com.google.code.gson:gson:2.8.6'
    compile 'io.gsonfire:gson-fire:1.8.3'
    compile 'io.swagger:swagger-annotations:1.6.0'
    compile 'com.google.code.findbugs:jsr305:3.0.2'
}

openApiGenerate {
    generatorName = "java"   // Language target
    inputSpec = "$rootDir/petstore-one.yaml".toString() // Path to specifications file
}
```

##### Steps to reproduce

When I implement the client use like this, I'd like to work with Error object from ApiException but just responseBody is returned but it is not deserialized :(
```java
                try {
			api.listPets(transaction);
		} catch (ApiException ex) {
			// How get Error object?
		}
```

I tried for several language (Ruby, JS, Python), it seems that there is the same issue.

### PR DESCRIPTION
**Pull request by Chhida**
# Pull Request Description

## Title
`[Java][jersey3] Add error entity deserialization to ApiException`

---

## Description

Add support for deserializing error response bodies into the model types defined in the OpenAPI specification for the jersey3 library.

## Problem

When an OpenAPI spec defines error response schemas (e.g., `ErreurApiWrapper` for 400 errors), these models are generated but not used by the generated client code. The `ApiException` only contains the raw response body as a string, requiring manual deserialization.

## Solution

- Add `errorEntity` field to `ApiException` to store the deserialized error object
- Add `getErrorEntity()` method to retrieve the deserialized error model
- Build error types map from API responses in each API method
- Deserialize error response body based on the HTTP status code

## Changes

- `apiException.mustache`: Add `errorEntity` field, constructor, and `getErrorEntity()` method
- `api.mustache`: Build `localVarErrorTypes` map from API responses  
- `ApiClient.mustache`: Add `deserializeErrorEntity` method and update `invokeAPI` signature

## Usage

```java
try {
    api.someOperation();
} catch (ApiException e) {
    Object errorEntity = e.getErrorEntity();
    if (errorEntity instanceof ErreurApiWrapper) {
        ErreurApiWrapper error = (ErreurApiWrapper) errorEntity;
        // Handle error
    }
    // Original response body still available
    String responseBody = e.getResponseBody();
}
```

## Testing

- All 634 Java tests passed (no regressions)
- Generated sample code verified to contain the new functionality

## Related Issues

- Fixes #4777 - ApiException does not contain Error object
- Related to PR #5609 (jersey2 implementation, closed)

## Checklist

- [x] Read the contribution guidelines
- [x] Run tests locally (634 Java tests passed)
- [x] Generated samples verified

---
## Summary by cubic
Adds automatic error model deserialization to `jersey3` clients so `ApiException` carries a typed error entity while keeping the raw response body. Regenerates samples and adds tests; fixes edge cases and build issues.

- **New Features**
  - `ApiException`: added transient `errorEntity` and `getErrorEntity()`.
  - `ApiClient`: added `deserializeErrorEntity`; buffers the entity; `invokeAPI` now accepts an `errorTypes` map.
  - Generated APIs: build and pass a status-code → `GenericType` map for error responses (`"0"` for default).

- **Bug Fixes**
  - Error handling: return null when deserialization fails; include all responses in the `errorTypes` map; mark `errorEntity` as transient.
  - Tests/samples: add unit/functional tests; set `hideGenerationTimestamp=true`; fix functional test template path and UTF-8 charset; add Javadoc to `deserializeErrorEntity`; regenerate `jersey3` samples and add missing deps (`jakarta.validation-api` as provided, `commons-lang3`, `org.tomitribe:tomitribe-http-signatures`); remove duplicate `junit-version`; regenerate again to fix sample compilation.

<sup>Written for commit 75b74189d538000fcf1e1a3b53cc3402ec19c514. Summary will update on new commits.</sup>

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by wing328** [comment]
please follow step 3 to update the samples as well so that CI can verify the change

**Comment by Chhida** [comment]
> please follow step 3 to update the samples as well so that CI can verify the change

Hi @wing328,

Thanks for the feedback!

I’ve updated the samples as requested (step 3), and all commits are now correctly linked to my GitHub account.

Everything should be ready for review.

Thanks!

**Comment by Chhida** [comment]
> > please follow step 3 to update the samples as well so that CI can verify the change
> 
> Hi @wing328,
> 
> Thanks for the feedback!
> 
> I’ve updated the samples as requested (step 3), and all commits are now correctly linked to my GitHub account.
> 
> Everything should be ready for review.
> 
> Thanks!



Hi @wing328,

I've verified that the KotlinReservedWordsTest failure is a pre-existing flaky test - it passes locally (101 tests, 0 failures). This is unrelated to the jersey3 errorEntity changes.

Could you please confirm this and proceed with the review?

**Comment by wing328** [comment]
cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

**Comment by wing328** [comment]
is it correct to say that you've been using this fix in your production environment and it has been working fine for your use cases?

**Comment by Chhida** [comment]
> is it correct to say that you've been using this fix in your production environment and it has been working fine for your use cases?

Hi @wing328,

Yes, the fix has been verified and is working correctly:

1. **Unit tests** - 642 Java tests pass with no regressions
2. **Integration tests** - 4 tests pass in the generated jersey3 sample, verifying:
   - errorEntity is correctly populated in ApiException
   - Backward compatibility (errorEntity is null when not provided)
   - Returns null on deserialization failure
   - Works with any object type

The functionality has been verified through tests and confirmed by our team.

We're waiting for approval of the PR so we can deploy it in our production environment.

File: modules/openapi-generator/src/main/resources/Java/libraries/jersey3/ApiClient.mustache [hunk 1]
```diff
@@ -1351,6 +1357,21 @@ public class ApiClient{{#jsr310}} extends JavaTimeFormatter{{/jsr310}} {
       }
     }
   }
+  
+  private Object deserializeErrorEntity(Map<String, GenericType> errorTypes, Response response) {
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/jersey3/ApiClient.mustache:1370]: please add a docstring explaining what this function does

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityFunctionalTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityFunctionalTest.java]: please remove line 3

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityTest.java]: please remove line 3

File: modules/openapi-generator/src/main/resources/Java/libraries/jersey3/ApiClient.mustache (hunk 1 above)
Comment (Chhida) [reply modules/openapi-generator/src/main/resources/Java/libraries/jersey3/ApiClient.mustache:1370]: done

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityFunctionalTest.java (hunk 2 above)
Comment (Chhida) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityFunctionalTest.java]: @wing328   done

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityTest.java (hunk 3 above)
Comment (Chhida) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/jersey3/JavaJersey3ErrorEntityTest.java]: done

**Comment by Chhida** [comment]
> @cubic-dev-ai review

@cubic-dev-ai review

**Comment by Chhida** [comment]
> @cubic-dev-ai review

@cubic-dev-ai review

**Comment by Chhida** [comment]
> **No issues found** across 107 files
> 
> **Note:** This PR contains a large number of files. cubic only reviews up to 75 files per PR, so some files may not have been reviewed. cubic prioritises the most important files to review.

@cubic-dev-ai review

**Comment by Chhida** [comment]
> You're iterating quickly on this pull request. To help protect your rate limits, cubic has paused automatic reviews on new pushes for now—when you're ready for another review, comment `@cubic-dev-ai review`.

@cubic-dev-ai review

**Comment by Chhida** [comment]
@cubic-dev-ai review

**Comment by Chhida** [comment]
@cubic-dev-ai review

**Comment by Chhida** [comment]
@cubic-dev-ai review

**Comment by Chhida** [comment]
@cubic-dev-ai review

**Comment by Chhida** [comment]
@cubic-dev-ai review

**Comment by Chhida** [comment]
> **11 issues found across 98 files (changes from recent commits).**
> 
> **Note:** This PR contains a large number of files. cubic only reviews up to 75 files per PR, so some files may not have been reviewed. cubic prioritises the most important files to review.
> 
> Prompt for AI agents (unresolved issues)
> ```
> 
> Check if these issues are valid — if so, understand the root cause of each and fix them. If appropriate, use sub-agents to investigate and fix each issue separately.
> 
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/Mammal.java">
> 
> <violation number="1">
> P2: Missing discriminator value will cause a NullPointerException at the switch statement and prevent the existing oneOf fallback deserialization from running.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/MapTest.java">
> 
> <violation number="1">
> P2: Using `EqualsBuilder.reflectionEquals` with `testRecursive=true` compares map internals instead of `Map.equals`, so two maps with identical entries but different internal layout can become unequal and produce inconsistent hash codes.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/ArrayOfArrayOfNumberOnly.java">
> 
> <violation number="1">
> P2: `EqualsBuilder.reflectionEquals(..., true)` performs recursive reflection into collection internals, so equality becomes implementation-detail dependent (e.g., ArrayList capacity/modCount) instead of content-based `List.equals`. This is a behavioral regression from the prior explicit field comparison and can make logically equal models compare unequal.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/Apple.java">
> 
> <violation number="1">
> P2: @Pattern uses Java regex syntax; the added JavaScript-style `/.../i` delimiters make the regex match literal slashes and a trailing `i`, so normal origin values won’t validate.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/MammalAnyof.java">
> 
> <violation number="1">
> P2: additionalProperties support is non-functional: custom serialization writes only the actual instance and deserialization never populates the map, so values added via putAdditionalProperty are silently dropped.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/ArrayTest.java">
> 
> <violation number="1">
> P2: Recursive reflection-based equals/hashCode changes equality semantics for collection fields (compares internal ArrayList state instead of List.equals), which can yield incorrect equality results vs the prior explicit field comparison.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/Dog.java">
> 
> <violation number="1">
> P2: `equals` uses recursive reflection while `hashCode` uses non-recursive object hash codes, which can violate equals/hashCode contract for arbitrary nested `additionalProperties` values.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/Cat.java">
> 
> <violation number="1">
> P2: Using recursive reflection-based equals on a model that now contains arbitrary additionalProperties can throw InaccessibleObjectException on Java 9+ (reflective access into JDK classes) and changes equality semantics away from explicit field comparison.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/MixedPropertiesAndAdditionalPropertiesClass.java">
> 
> <violation number="1">
> P2: Using EqualsBuilder.reflectionEquals with recursive reflection can throw InaccessibleObjectException on Java 9+ when it reflects into JDK types (UUID/OffsetDateTime/HashMap), and it changes equality semantics versus explicit field comparisons.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/BananaReq.java">
> 
> <violation number="1">
> P2: Reflection-based equals with recursive comparison can throw InaccessibleObjectException on JDK 9+ module encapsulation and is a behavioral regression from explicit field comparison. Consider reverting to explicit Objects.equals to avoid reflective access into JDK types like BigDecimal.</violation>
> </file>
> 
> <file name="samples/client/petstore/java/jersey3/src/main/java/org/openapitools/client/model/OuterComposite.java">
> 
> <violation number="1">
> P2: equals now uses recursive reflection (testRecursive=true), which changes equality semantics and can cause reflective access failures on JDK types like BigDecimal under Java 9+ module encapsulation. This is a regression from the previous explicit field comparisons.</violation>
> </file>
> ```
> 
> Reply with feedback, questions, or to request a fix. Tag `@cubic-dev-ai` to re-run a review.


@wing328   These 11 P2 issues reported (Mammal, MapTest, ArrayOfArrayOfNumberOnly, Apple, MammalAnyof, ArrayTest, Dog, Cat, MixedPropertiesAndAdditionalPropertiesClass, BananaReq, OuterComposite) concern pre-existing models in the generated samples - are these issues related to PR #23542 (error entity deserialization) or are they pre-existing regressions in the Jersey3 templates?

PR #23542 only modifies:
- `apiException.mustache` - added errorEntity field and getErrorEntity() method
- `ApiClient.mustache` - added deserializeErrorEntity() method
- `api.mustache` - added localVarErrorTypes map

These reported issues appear to be related to:
1. EqualsBuilder.reflectionEquals with recursive reflection (MapTest, ArrayOfArrayOfNumberOnly, ArrayTest, Dog, Cat, MixedPropertiesAndAdditionalPropertiesClass, BananaReq, OuterComposite)
2. @Pattern regex syntax (Apple)
3. additionalProperties serialization (MammalAnyof)
4. Discriminator null handling (Mammal)

Please confirm whether these issues need to be fixed in this PR or if they are out of scope (pre-existing template issues).

**Comment by wing328** [comment]
just merged your contribution. thank you

other reviews by cubic-dev-ai can be done in separate PRs instead.
